dashboard/app/dsl/level_group_dsl.rb

```diff
@@ -28,6 +29,22 @@ def page
     @hash[:pages] << {levels: @current_page_level_names}
   end
 
+  def text(name)
+    # Ensure level is appropriate type.
+    level = Level.where(name: name).first # For some reason find_by_name doesn't always work here!
```

Comment:
Unfortunately I don't think we figured out why, though both @islemaster and @laurelfan investigated around this time.  This PR is probably relevant reading: https://github.com/code-dot-org/code-dot-org/pull/7844